python_modules/dagster/dagster_tests/core_tests/asset_defs_tests/test_assets.py

```diff
@@ -458,7 +459,9 @@ def my_asset(context):
         yield Output(1, "key1")
         yield Output(2, "key2")
 
-    materialize_to_memory([my_asset])
+    with instance_for_test() as instance:
+        result = materialize([my_asset], instance=instance)
+        assert result.success
```

Comment:
this feels unfortunate to me, and is a result of us explicitly erroring if resource defs are directly provided.